packages/office-ui-fabric-react/src/components/Fabric/Fabric.base.tsx

```diff
@@ -38,10 +36,32 @@ export class FabricBase extends React.Component<
     if (win) {
       this._disposables.push(on(win, 'mousedown', this._onMouseDown, true), on(win, 'keydown', this._onKeyDown, true));
     }
+
+    const classNames = this._getClassNamesHelper();
+    if (this.props.applyThemeToBody) {
+      const currentDoc: Document = getDocument(this._rootElement.current)!;
+      currentDoc.body.classList.add(classNames.bodyThemed);
+    }
   }
 
   public componentWillUnmount(): void {
     this._disposables.forEach((dispose: () => void) => dispose());
+
+    const classNames = this._getClassNamesHelper();
+    if (this.props.applyThemeToBody) {
+      const currentDoc: Document = getDocument(this._rootElement.current)!;
+      currentDoc.body.classList.remove(classNames.bodyThemed);
+    }
+  }
+
+  private _getClassNamesHelper(): IProcessedStyleSet<IFabricStyles> {
+    const { className } = this.props;
+    const classNames = getClassNames(getStyles, {
+      theme: this.props.theme!,
```

Comment:
Should check the return value of `this.props.theme`, not use `!`.

Reply:
done.

Reply:
I tried to avoid using this ! but the only way to to do an if check for `this.props.theme` is outside getClassNames call (unless I'm forgetting something?) and when I do that, the method needs an alternate return value besides `IProcessedStyleSet<IFabricStyles>` like undefined but adding that to this method signature is introducing unnecessary ambiguity especially because I looked at our usage of ! inside getClassNames calls across the Fabric codebase and didn't find any other cases where it was avoided.